File: 121_BTBSS/2ms.java
class Solution {
  public int maxProfit(int[] prices) {
    int buyingPrice = prices[0], maxProfit = 0;

    for(int i = 1; i < prices.length; i++) {
      if(prices[i] < buyingPrice)
        buyingPrice = prices[i];
      else if (prices[i] - buyingPrice > maxProfit)
        maxProfit = prices[i] - buyingPrice;
    }

    return maxProfit;
  }
}

File: 2706_Buy_Two_Chocolates/2ms.ex
defmodule Solution do
  @spec buy_choco(prices :: [integer], money :: integer) :: integer
  def buy_choco(prices, money) do
    {min1, rest} = pop_min(prices)
    {min2, _} = pop_min(rest)
    min_bill = min1 + min2
    if min_bill <= money, do: money - min_bill, else: money
  end

  def pop_min(prices) do
    min = Enum.min(prices)
    {min, List.delete(prices, min)}
  end
end
